pkg/ebpf/events_pipeline.go

```diff
@@ -220,16 +221,20 @@ func (t *Tracee) decodeEvents(outerCtx context.Context, sourceChan chan []byte)
 				PodSandbox:          containerInfo.Pod.Sandbox,
 				EventID:             int(ctx.EventID),
 				EventName:           eventDefinition.Name,
+				MatchedScopes:       ctx.MatchedScopes,
 				ArgsNum:             int(ctx.Argnum),
 				ReturnValue:         int(ctx.Retval),
 				Args:                args,
 				StackAddresses:      StackAddresses,
 				ContextFlags:        parseContextFlags(ctx.Flags),
 			}
 
-			if !t.shouldProcessEvent(evt) {
-				t.stats.EventsFiltered.Increment()
-				continue
+			// base events for derived ones should be filtered in later stage
+			if _, ok := t.eventDerivations[eventId]; !ok {
+				if !t.shouldProcessEvent(&evt) {
+					t.stats.EventsFiltered.Increment()
```

Comment:
Isn't it a problem that we don't increment this counter later when calculating derived events (in case it shouldn't be processed)?

Reply:
Raised a commit for that. 👍🏼